apps/vmq_server/src/vmq_http_mgmt_api.erl

```diff
@@ -0,0 +1,159 @@
+%% Copyright 2016 Erlio GmbH Basel Switzerland (http://erl.io)
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(vmq_http_mgmt_api).
+-behaviour(vmq_http_config).
+-export([init/3,
+         rest_init/2,
+         allowed_methods/2,
+         content_types_provided/2,
+         options/2,
+         is_authorized/2,
+         malformed_request/2,
+         to_json/2,
+         to_html/2]).
+
+-export([routes/0,
+         create_api_key/0,
+         delete_api_key/1,
+         list_api_keys/0]).
+
+-define(ENV_API_KEYS, http_mgmt_api_keys).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% Cowboy REST Handler
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+init(_Transport, _Req, _Opts) ->
+    {upgrade, protocol, cowboy_rest}.
+
+rest_init(Req, _Opts) ->
+    {ok, Req, undefined}.
+
+allowed_methods(Req, State) ->
+    {[<<"POST">>, <<"GET">>, <<"OPTIONS">>, <<"HEAD">>], Req, State}.
+
+content_types_provided(Req, State) ->
+    {[{<<"application/json">>, to_json},
+      {<<"text/html">>, to_html}], Req, State}.
```

Comment:
Why the second content type. Won't all the content returned be strictly JSON? As the `to_html/2` function just calls the `to_json/2` function I suppose everything really will be JSON and not HTML?